examples/simple_saga/src/aggregate.rs

```diff
@@ -1,79 +1,80 @@
 use async_trait::async_trait;
 use sqlx::{Pool, Postgres};
+use std::sync::Arc;
+use uuid::Uuid;
 
-use esrs::aggregate::{Aggregate, AggregateManager, AggregateState};
-use esrs::policy::PgPolicy;
-use esrs::projector::PgProjector;
-use esrs::store::{PgStore, StoreEvent};
+use esrs::postgres::PgStore;
+use esrs::{Aggregate, AggregateManager, AggregateState, Policy, StoreEvent};
 
 use crate::structs::{LoggingCommand, LoggingError, LoggingEvent};
 
-// A store of events
-pub type LogStore = PgStore<LoggingEvent, LoggingError>;
-
+#[derive(Clone)]
 pub struct LoggingAggregate {
-    event_store: LogStore,
+    event_store: Arc<PgStore<Self>>,
 }
 
 impl LoggingAggregate {
     pub async fn new(pool: &Pool<Postgres>) -> Result<Self, LoggingError> {
-        Ok(Self {
-            event_store: Self::new_store(pool).await?,
-        })
-    }
-
-    async fn new_store(pool: &Pool<Postgres>) -> Result<LogStore, LoggingError> {
-        let projectors: Vec<Box<dyn PgProjector<LoggingEvent, LoggingError> + Send + Sync>> = vec![]; // There are no projections here
+        let event_store: PgStore<LoggingAggregate> = PgStore::new(pool.clone()).setup().await?;
 
-        let policies: Vec<Box<dyn PgPolicy<LoggingEvent, LoggingError> + Send + Sync>> =
-            vec![Box::new(LoggingPolicy {})];
+        let this: Self = Self {
+            // This clone is cheap being that the internal fields of the store are behind an Arc
+            event_store: Arc::new(event_store.clone()),
```

Comment:
Since now `PgStore` is a smart pointer we can simplify the example
```suggestion
            event_store: event_store.clone(),
```